api/config/rolepermission.php

```diff
@@ -0,0 +1,521 @@
+    |
+    | A map of all resources that a specific action can be taken on.
+    |
+    */
+    'resources' => [
+        'clss' => 'classification',
+        'dpt' => 'department',
+        'sk' => 'skill',
+        'sf' => 'skillFamily',
+        'usr' => 'user',
+        'ubi' => 'userBasicInfo',
+        'pl' => 'pool',
+        'ppa' => 'publishedPoolAdvertisement',
+        'dp' => 'draftPool',
+        'pcd' => 'poolClosingDate',
+        'app' => 'application',
+        'sapp' => 'submittedApplication',
+        'aprf' => 'applicantProfile',
+        'dapp' => 'draftApplication',
+        'apps' => 'applicationStatus',
+        'ac' => 'applicantCount',
+        'sr' => 'searchRequest',
+        't' => 'team',
+        'tusrbi' => 'teamUsersBasicInfo',
+        'rle' => 'role'
+    ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Permissions
+    |--------------------------------------------------------------------------
+    |
+    | A map of all permissions with $name => $display_name shape.
+    |
+    */
+    'permissions' => [
+        'view-any-classification' => [
+            'en' => 'View Any Classification',
+            'fr' => 'Afficher toute classification'
+        ],
+        'create-any-classification' => [
+            'en' => 'Create Any Classification',
+            'fr' => 'Créer n\'importe quelle classification'
+        ],
+        'update-any-classification' => [
+            'en' => 'Update Any Classification',
+            'fr' => 'Mise à jour de toute classification'
+        ],
+        'delete-any-classification' => [
+            'en' => 'Delete Any Classification',
+            'fr' => 'Supprimer toute classification'
+        ],
+
+        'view-any-department' => [
+            'en' => 'View Any Department',
+            'fr' => 'Voir n\'importe quel département'
+        ],
+        'create-any-department' => [
+            'en' => 'Create Any Department',
+            'fr' => 'Créer n\'importe quel département'
+        ],
+        'update-any-department' => [
+            'en' => 'Update Any Department',
+            'fr' => 'Mise à jour de n\'importe quel département'
+        ],
+        'delete-any-department' => [
+            'en' => 'Delete Any Department',
+            'fr' => 'Supprimer un département'
+        ],
+
+        'view-any-skill' => [
+            'en' => 'View Any Skill',
+            'fr' => 'Afficher n\'importe quelle compétence'
+        ],
+        'create-any-skill' => [
+            'en' => 'Create Any Skill',
+            'fr' => 'Créer n\'importe quelle compétence'
+        ],
+        'update-any-skill' => [
+            'en' => 'Update Any Skill',
+            'fr' => 'Mettre à jour n\'importe quelle compétence'
+        ],
+        'delete-any-skill' => [
+            'en' => 'Delete Any Skill',
+            'fr' => 'Supprimer toute compétence'
+        ],
+
+        'view-any-skillFamily' => [
```

Comment:
Is there a specific reason why a combination of cases (kebab-case and camelCase) is used for these keys instead of just kebab-case?

Reply:
This is to avoid confusion. We have the pattern of `{action}-{scope}-{resource}`. So the hyphens are acting like spaces between camel case names. It just so happens that resources are the only multi-word names that require camel case.

Reply:
It clearly demarcates the three sections (action, scope, resource) of each permission